Gui adaptions

MADITT opened the pull request:
Gui improvements according to previous lecture and current issues

MADITT committed:
+Update: Rescale elevator

MADITT committed:
Merge branch 'rmi_connection_tests' into gui_adaptions

MADITT committed:
+Fix: Visibility Bug +Update: Resize Listener

MADITT committed:
+Update: prevent elevator glitching out if no connection is set up

MADITT committed:
Merge remote-tracking branch 'origin/master' into gui_adaptions

MADITT committed:
+Update: "Beautify" Symbols

MADITT committed:
+Update: Padding of state content

MADITT committed:
+Fix: Direction not visible if current floor equal to target floor

MADITT committed:
+Fix: Close Door background issue

MADITT commented:
@skaupper @pagdot this pr should fix the three following issues:
#10 #11 #14
Did I understand the issues correctly or is still something missing?

skaupper commented on src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java:
```diff
@@ -389,12 +392,21 @@ private void translateElevator(Integer percentage) {
         private final BooleanProperty isServiced = new SimpleBooleanProperty(false);
     }
+    
+    public void resizeElevator() {
```
Isn't that pretty much the same as the translate method?

MADITT replied:
True this is still ongoing and should be refactored soon.

MADITT replied:
Is refactored.

skaupper commented on src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java:
```diff
@@ -326,9 +329,9 @@ public void initialize(URL location, ResourceBundle resources) {
         ivDoorStateOpen.visibleProperty().bind(isDoorOpen.and(anyElevatorSelected).and(isConnected));
 
-        ivGElvDirUp.visibleProperty().bind(isDirectionUp.and(anyElevatorSelected));
+        ivGElvDirUp.visibleProperty().bind(isDirectionUp.and(anyElevatorSelected).and(isOnTargetFloor.not()));
```
That's not exactly what I meant with #11. The images should be disabled if the committed direction provided by the model equals `Uncommitted`. Right now the model does provide that field but does not allow to set it manual (it is automatically set when `setTargetFloor` is called). So, please add a mechanism to monitor that property. Unfortunately you cannot easily test it by now.

MADITT replied:
Is added now in the intended way.

skaupper commented on src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECC.java:
```diff
@@ -51,6 +51,11 @@ public void start(Stage stage) throws Exception {
         stage.setTitle(Messages.getString("ECC.2"));
         stage.setScene(scene);
+        
+        stage.heightProperty().addListener((obs, oldVal, newVal) -> {
```
Can't that be moved to the controller itself? In case it isn't triggered after startup you could call it manually during initialization or something.

MADITT replied:
I refactored the controller in a way, that it can set the listener itself but it still needs the stage to do so. But should be a cleaner and more adaptable approach now.

skaupper reviewed (commented):
I like the updated icons and layout!

Some small things concerning the code and #11 should be fixed, though.

skaupper commented:
Your changes to the visibility of the direction icons have broken the unit tests btw.

MADITT committed:
+Fix: implemented uncommited direction the right way

MADITT committed:
+Update: Refactored resize listener

skaupper commented on src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java:
```diff
@@ -383,18 +389,31 @@ protected void gotoTargetFloor(ActionEvent event) {
     }
+    
+    public void setStageAndSetUpListeners(Stage mainStage) {
+    	stage = mainStage;
```
Do we need the stage as a member?

MADITT replied:
Checked it is not necessary as tests showed.

skaupper commented on src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java:
```diff
@@ -383,18 +389,31 @@ protected void gotoTargetFloor(ActionEvent event) {
 
-    private void translateElevator(Integer percentage) {
+    private void translateElevator(Integer percentage, Boolean triggerResize) {
+    	if(triggerResize) {
```
Well, that is ugly af. An overload without parameters which translates the elevator to `oldPercentage` would do, I guess.

skaupper replied:
And btw, formatting!

MADITT replied:
Why u always have so good architectural thoughts. Changed it accordingly. 

Formatted it now by hand hope it is any better?

skaupper commented on src/main/resources/at/fhhagenberg/esd/sqe/ws20/ECCMainGUI.fxml:
```diff
@@ -144,31 +144,46 @@
 						    		<Label text="%ECC.8" />
-						    		<Label fx:id="lWeight" />
+						    		<Label fx:id="lWeight">
+                                 <padding>
```
Can you autoformat this file with eclipse?

MADITT replied:
done

MADITT committed:
-Update: Removed stage from controller +Update: Refactor resize

pagdot committed:
Clear rows on (re)connect

Fixes issue with too large rectangle in ui

pagdot commented on src/main/java/at/fhhagenberg/esd/sqe/ws20/gui/ECCController.java:
```diff
@@ -382,19 +388,30 @@ protected void gotoTargetFloor(ActionEvent event) {
     }
+    
+    public void setStageAndSetUpListeners(Stage mainStage) {
+    	mainStage.heightProperty().addListener((obs, oldVal, newVal) -> {
```
we should fix the codesmell here:

```c++
heightProperty().addListener((obs, oldVal, newVal) -> translateElevator());
```

MADITT replied:
Which code smell sry I don't get it.

MADITT replied:
Ah i see u saw it on Sonar Cloud. Should be fixed now.

MADITT committed:
+Update: Direction Lable can now handle all directions and effects gui

skaupper committed:
update ModelUpdateTest to check for uncommitted directions too

skaupper committed:
fix combobox selection in ECCPageObject

MADITT committed:
+Fix: Spelling mistake

MADITT committed:
+Fix: Code smell

skaupper committed:
make synchronization failures ConnectionErrors
add a separate setCommittedDirection method
set committed direction together with target floor
set committed direction to uncommitted if the elevator is at its destination

skaupper committed:
Merge branch 'master' into gui_adaptions

skaupper committed:
fix code smells

skaupper committed:
verify that exceptions other than ECCError and RemoteException lead to an IllegalStateException during a call to the model

skaupper commented:
The code smell concerning `ElevatorImplTest` will be fixed with #21 after this PR is merged.

MADITT committed:
+Fix: Padding

pagdot committed:
refactor: fix variable name

pagdot committed:
Improve connection handling

fix reconnect if initial connection failed

pagdot committed:
Refactor: Fix code smells
